firebase-installations/src/main/java/com/google/firebase/installations/local/PersistedInstallation.java

```diff
@@ -164,6 +169,7 @@ public PersistedInstallationEntry insertOrUpdatePersistedInstallationEntry(
       // There isn't a lot we can do when this happens, other than crash the process. It is a
       // bit nicer to eat the error and hope that the user clears some storage space on their
       // device.
+      Log.e(TAG, "error encountered while persisting firebase installation." + e);
```

Comment:
String.format(Error while writing to disc (%s): %s", filename, exception)

If there is always only 1 file, you can omit the filename.
Instead of "disc" we could also use "shared pref".